pkg/controllers/kdd-ipam/controller.go

```diff
@@ -0,0 +1,228 @@
+// Copyright (c) 2017 Tigera, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kddipam
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	uruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+
+	"github.com/projectcalico/kube-controllers/pkg/config"
+	"github.com/projectcalico/kube-controllers/pkg/controllers/controller"
+	bapi "github.com/projectcalico/libcalico-go/lib/backend/api"
+	"github.com/projectcalico/libcalico-go/lib/backend/model"
+	client "github.com/projectcalico/libcalico-go/lib/clientv3"
+)
+
+// Controller implements the Controller interface.  It is responsible for monitoring
+// kubernetes nodes and responding to delete events by removing them from the Calico datastore.
+type Controller struct {
+	ctx       context.Context
+	informer  cache.Controller
+	indexer   cache.Indexer
+	client    client.Interface
+	clientset *kubernetes.Clientset
+	bc        bapi.Client
+}
+
+type allocation struct {
```

Comment:
Could we use a very similarly shaped `AllocationAttribute` struct defined in the `model` package instead of creating a new one?